magenta/models/rl_tuner/README.md

```diff
@@ -0,0 +1,107 @@
+from data.
+
+The music theory reward functions are designed to constrain the actions of the
+network so that it chooses notes in accordance with a musical structure; for
+example, choosing harmonious interval steps and playing notes within the same 
+key. Several reward functions have been written, but these could easily be 
+improved and extended!
+
+In addition to the normal Q function, this code provides the ability to train 
+the network with the [Psi learning][psi learning] and [G learning][g learning]
+functions, which can be set with the `algorithm` hyperparameter. For details 
+on each algorithm, see [our paper][our arxiv].
+
+## Code structure
+* 	In the constructor, RLTuner loads the `q_network`, `target_q_network`, and 
+	`reward_rnn` from a checkpointed Note RNN.
+
+*	The tensorflow graph architecture is defined in the `build_graph` 
+	function.
+
+*	The model is trained using the `train` function. It will continuously 
+	place notes by calling `action`, receive rewards using `collect_reward`, 
+	and save these experiences using `store`.
+
+*	The network weights are updated using `training_step`, which samples 
+	minibatches of experience from the model's `experience` buffer and uses 
+	this to compute gradients based on the loss function in `build_graph`.
+
+*	During training, the function `evaluate_model` is occasionally run to 
+	test how much reward the model receives from both the Reward RNN and the 
+	music theory functions.
+
+*	After the model is trained, you can use the `save_model_and_figs` function
+	to save a checkpoint of the model and a set of figures of the rewards over 
+	time. 
+
+*	Finally, use `generate_music_sequence` to generate a melody with your 
+	trained model! You can also call this function before training, to see how 
+	the model's songs have improved with training! If you set the 
+	`visualize_probs` parameter to *True*, it will also plot the 
+	note probabilities of the model over time.
+
+## Running the code
+To train the model you can use the provided jupyter notebook ([rl_tuner.ipynb][ipynb]), or, you can simply run:
+
+```
+bazel run magenta/models/rl_tuner:rl_tuner_train 
+```
+
+## Tuning your own model
+
+By default, if you don't provide a Note RNN checkpoint file to load, the code 
+will automatically download and use the checkpointed model we used for 
+[our paper][our arxiv] from [here][note rnn ckpt].
+
+If you want to use your own model, you need to pass in the directory containing
+it using the `note_rnn_checkpoint_dir`, and the hyperparameters you used to 
+train it via `note_rnn_hparams`. You can also pass in a path to the checkpoint
+file directly using `backup_checkpoint_file`.
+
+We also support tuning a *basic_rnn* trained using the Magenta code! To tune
+a basic_rnn, use the same `note_rnn_checkpoint_dir` parameter, but set the 
+`note_rnn_type` parameter to 'basic_rnn'. We also provide the script 
+`unpack_ckpt_from_bundle` to help you extract a checkpoint file from one of the 
```

Comment:
I think having a script like this is handy, but can the model just read the bundle directly?